Gruntfile.js

```diff
@@ -1099,58 +988,314 @@ module.exports = function(grunt) {
     'copy:libsForSimpleSocksNode'
   ]);
   registerTask(grunt, 'uprobe', [
-    'base', 
-    'browserify:uprobeFreedomModule', 
-    'copy:libsForUprobeChromeApp', 
+    'base',
+    'browserify:uprobeFreedomModule',
+    'copy:libsForUprobeChromeApp',
     'copy:libsForUprobeFirefoxApp'
   ]);
   registerTask(grunt, 'zork', [
     'base', 
-    'browserify:zorkFreedomModule', 
-    'copy:libsForZorkChromeApp', 
-    'copy:libsForZorkFirefoxApp', 
-    'copy:libsForZorkNode', 
+    'browserify:zorkFreedomModule',
+    'copy:libsForZorkChromeApp',
+    'copy:libsForZorkFirefoxApp',
+    'copy:libsForZorkNode',
     'exec:installFreedomForNodeForZork'
   ]);
-  registerTask(grunt, 'version_file', [
-    'gitinfo',
-    'string-replace:version'
+
+  // Chrome
+  registerTask(grunt, 'build_chrome', [
+    'build_chrome_app',
+    'build_chrome_ext'
   ]);
+
+  // =========================================================================
+  // Chrome App
+  // =========================================================================
+
   registerTask(grunt, 'build_chrome_app', [
     'base',
-    'copy:chrome_app'
+    'chromeAppMainCoreEnv',
+    'chromeAppExtMissing',
+    'copy:chromeApp',
+  ]);
+  registerTask(grunt, 'chromeAppMainCoreEnv',
+      'Builds build/dist/chrome/app/scripts/main.core-env.static.js', [
+    'compileTypescript',
+    'browserify:chromeAppMainCoreEnv'
+  ]);
+  registerTask(grunt, 'chromeAppExtMissing',
+      'Builds build/src/chrome/app/polymer/vulcanized.{html,js}', [
+    'compileTypescript',
+    'copy:resources',
+    'copy:chromeAppBower',
   ].concat(fullyVulcanize('chrome/app/polymer', 'ext-missing', 'vulcanized')));
+  
+  grunt.config.merge({
+    browserify: {
+      chromeAppMainCoreEnv: Rule.browserify('chrome/app/scripts/main.core-env'),
+    },
+    copy: {
+      chromeApp: Rule.copyLibs({
+        npmLibNames: ['freedom-for-chrome', 'forge-min'],
+        pathsFromDevBuild: [
+          'generic_core'
+        ].concat(backendFreedomModulePaths),
+        pathsFromThirdPartyBuild: backendThirdPartyBuildPaths,
+        files: getExtraFilesForCoreBuild(chromeAppDevPath).concat({
+          expand: true, cwd: 'src/',
+          src: ['icons/128_online.png', 'fonts/*'],
+          dest: chromeAppDevPath
+        }),
+        localDestPath: 'chrome/app/'
+      }),
+      chromeAppBower: {
+        expand: true, cwd: 'third_party',
+        src: ['bower/**/*'],
+        dest: chromeAppDevPath
+      },
+    }
+  });
+
+  // =========================================================================
+  // Chrome Extension
+  // =========================================================================
+
   registerTask(grunt, 'build_chrome_ext', [
     'base',
```

Comment:
I don't think you need `base` anymore - the `chromeExt*` targets each only depend on `compileTypescript`. I quickly tried this and the extension seems to build just fine.